added install script

johncoleman83 opened the pull request:
Based on the question, 

> "what causes someone who tries the editor to stop using it?" 

and based on issue #35, I thought that if I could open the editor from any directory I would be much more likely to continue to use it.  If this feature already exists, I was not able to understand how, and so I created an install script that creates a new command in the `$PATH`.  The command is `cedt`, but this can be updated to whatever the user wants.  The installation directory (the repo) is copied or updated to the path: `/opt/ci_edit`, and the command is a sym link to the executable `ci.py`.  The command is stored in the `$PATH` dir: `/usr/local/bin`, which is designated for user programs not managed by the distribution package manager.

If you want this concept, and want an upgrade, I can help to create a Debian or Python package.

dschuyler commented:
Thanks! It will be a few days before I can look over the PR. Making ci_edit easier to get started with and use regularly is really appreciated.

dschuyler commented on readme.md:
````diff
@@ -5,6 +5,32 @@ ci_edit is a text editor. It can help you view or edit text files.
+$ cedt README.md
+```
+
 ## What ci_edit can do for you
````
Let's do the "What ci_edit can do for you" section just above the Installation/Usage.

johncoleman83 commented:
I changed the order of the sections of the readme.md so that the "**What ci_edit can do for you**" section comes before **installation** of shell command section.  LMK, I can put the **installation** section anywhere.

dschuyler commented on .gitignore:
```diff
@@ -1,2 +1,4 @@
\ No newline at end of file
+*.pyo
+*~
+\#*
```
I recognize the other two, but just out of curiosity, what generates \#* files?

johncoleman83 replied:
`emacs` buffer files when killed

johncoleman83 replied:
I might add `__pycache__` if I start working on python code, I usually don't create `#*` files, but added it just to be safe while I edited files (it's not a necessary update).

dschuyler commented on readme.md:
````diff
@@ -32,8 +33,35 @@ a start. It's has the necessary features of a basic text editor and a few fancy
+* This command opens the text editor from any directory
+
+```
+$ cedt
````
What are the advantages for using the name cedt? I suppose we can avoid typing all those pesky 'i's :)

How about a command name like:
- ci
- ciedit
- ci_edit
- edit
- ce

My concern with cedt is that it's an awkward thing to type on a qwerty keyboard. The middle finger hits all three rows, followed by a reach for the 't'.

Trivia: the shortcut I use for ci_edit is 'we' for no other reasons than it's easy and fast to type; and didn't conflict with another command. I'd given some thought to changing the name of ci_edit to we_edit, but thought ci_edit was more distinct when spoken.

johncoleman83 replied:
Totes agree, I like the idea to use an Environmental Variable.

dschuyler commented on install.sh:
```diff
@@ -0,0 +1,39 @@
+echo "*                                                             *"
+echo "*   installation copies (or UPDATES) current directory to:    *"
+echo "*                  the path:  /opt/${APP_DIR}                    *"
+echo "*   and creates the (or updates) the command 'cedt' to:       *"
```
How about using $CMD in place of cedt here

johncoleman83 replied:
great idea, I'll work on it.

dschuyler commented on install.sh:
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
```
Please add license (it may be easier to copy paste from one of the .py source files than from here):
```
# Copyright 2017 Google Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
```

johncoleman83 replied:
will do

johncoleman83 replied:
completed

dschuyler commented on install.sh:
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+# init script that creates an executable in /usr/local/bin
+# so that ci_edit can be used from any directory
```
Maybe a slight re-wording to say that it copies the files to /opt and adds a sym-link in /usr/local/bin?

johncoleman83 replied:
sounds good!

dschuyler commented on install.sh:
```diff
@@ -0,0 +1,39 @@
+echo "*   and creates the (or updates) the command 'cedt' to:       *"
+echo "*                   the path: /usr/local/bin                  *"
+echo "*                                                             *"
+echo "* *********************************************************** *"
```
Nice user explanation.

johncoleman83 commented:
I heavily updated the documentation in README, install comments, and install user directions:

- added license to install file
- added option for user to customize the execution command
- changed default execution command to `we`.  I like this as it is unique, short, and easy to type.

I can make any other requested updates, just LMK

johncoleman83 commented:
@dschuyler, I like the changes you made to safe guard the `rm -rf` command, and all the comments you added, but I wonder, did you intend to make the install script copy all contents of `ci_edit` to directory `/opt/`?  When the install script is run like `./install.sh`, `$0` is `./install.sh` and so this is a relative path, and thus the install directory becomes `/opt/./`, which becomes a bit redundant and also doesn't organize the package neatly into it's own directory...  Originally, it was setup to copy the application to the directory: `/opt/ci_edit/`, which is a bit cleaner as ci_edit would have it's own separate directory for it's package.  Generally `/opt/` is supposed to have the packages in their own subdirectory.

LMK if you didn't intend this and I can fix it.
